hash.py

```diff
@@ -12,9 +12,12 @@ def __init__(self, tablesize=1024):
 
     def hash(self, key):
         hash_val = 0
-        for char in key:
-            hash_val += ord(char)
-        return hash_val % len(self.table)
+        try:
+            for char in key:
+                hash_val += ord(char)
+            return hash_val % len(self.table)
+        except:
+            raise AttributeError('Please input a valid string')
```

Comment:
As per the Py2 docs, an AttributeError is to be raised when an attribute reference or assignment fails.

While an AttributeError could conceivably occur here, it's also easily possible for a non-string to be handed to ord(), and probably more likely when someone is trying to use the HashTable. This would raise a different kind of error, a kind that might be better addressed by the message in your raised error than an AttributeError would be.

Additionally, this failure mode is not covered by your test_hash.py file.